### PR TITLE
Solve potential buffer overflow bugs

### PR DESCRIPTION
**Pull request by evintila**
Added checks to prevent out-of-bounds writings and unwanted memcpy ranges.

### CODE REVIEW

File: src/common/Thread.h [hunk 1]
```diff
@@ -25,12 +25,14 @@ class Thread
 	public:
 		Thread(ThreadFunc threadFunction, const char *threadName="") : exitFlag(false), thread_created(false), f(threadFunction)
 		{
+			size_t threadName_len = strlen(threadName);
```
Comment (ogasser) [inline src/common/Thread.h]: Should be called `threadNameLen`

File: src/common/Thread.h [hunk 2]
```diff
@@ -25,12 +25,14 @@ class Thread
...
 			// truncate to 15 chars + NULL byte
-			if (strlen(threadName) > THREAD_NAME_LENGTH) {
```
Comment (ogasser) [inline src/common/Thread.h]: And let's rename the constant to the better name `MAX_THREAD_NAME_LEN`

File: src/common/ipfixlolib/ipfixlolib.c [hunk 3]
```diff
@@ -2366,7 +2374,6 @@ int ipfix_delete_data_fields_upto_marker(ipfix_exporter *exporter)
                 msg(MSG_ERROR, "delete_data_fields_upto_marker called but there is no set.");
                 goto out;
         }
-
```
Comment (ogasser) [inline src/common/ipfixlolib/ipfixlolib.c]: unnecessary whitespace change

File: src/modules/analysis/fpsigmatcher/Signature.c [hunk 4]
```diff
@@ -235,8 +235,10 @@ void destruct_token(struct Token * token){
 }
 
 void add_Token(struct BayesSignature * signature, struct Token * token){
-	signature->tokens[signature->numOfTokens] = token;
-	signature->numOfTokens++;
+	if (signature->numOfTokens < signature->maxNumOfTokens) {
+		signature->tokens[signature->numOfTokens] = token;
+		signature->numOfTokens++;
+	}
```
Comment (ogasser) [inline src/modules/analysis/fpsigmatcher/Signature.c]: Should we log an error if the `else` case?

**Comment by nickbroon** [comment]
msg should be available for C code. Just need to include msg.h
https://github.com/tumi8/vermont/blob/master/src/common/ipfixlolib/ipfixlolib.c for example make heavy use of it.

**Pull request merged**